astroML/plotting/tools.py

```diff
@@ -129,17 +129,17 @@ def discretize_cmap(cmap, N):
     for key in ('red', 'green', 'blue'):
         # Find the N colors
         D = np.array(cdict[key])
-        I = interpolate.interp1d(D[:, 0], D[:, 1])
-        colors = I(colors_i)
+        interp = interpolate.interp1d(D[:, 0], D[:, 1])
+        colors = interp(colors_i)
         # Place these colors at the correct indices.
         A = np.zeros((N + 1, 3), float)
         A[:, 0] = indices
         A[1:, 1] = colors
         A[:-1, 2] = colors
         # Create a tuple for the dictionary.
         L = []
-        for l in A:
-            L.append(tuple(l))
+        for c in A:
```

Comment:
I'm not sure I see why this is better, maybe call the single letter `color`?

Reply:
Flake8 will complain about ambiguous variable names for I, l, 1, o, 0. Used `c` to match the coding style of file. Renamed to color.